### ISSUE TITLE
Extend data-plane-azure-data-factory extension to support data transfer from blob storage to Dataverse

### ISSUE THREAD

**Issue by huanji-sun-007** [eclipse-edc/Connector#2705]
# Feature Request
Currently, [data-plane-azure-data-factory](https://github.com/eclipse-edc/Connector/tree/main/extensions/data-plane/data-plane-azure-data-factory) extension only supports Azure blob as both storage and destination.
We can also support other connectors such as Blob (as source) to Dataverse(as destination), to allow data transfer from blob to Dataverse. 

I am doing the implementation now under this [PR](https://github.com/huanji-sun-007/Connector/pull/1/files) in the forked repo, and I want to confirm if this work is valuable and get feedback from you in advance through this issue. If there are no problems I will complete the work and send the PR then.

## Which Areas Would Be Affected?
`data-plane-azure-data-factory` extension

## Why Is the Feature Desired?

- In our previous project, we had a scenario where data needed to be transferred from blob storage to Microsoft Dataverse. It would be helpful to have support for this.
- One of the benefits of using Data Factory is its ability to support over 80 [connectors](https://learn.microsoft.com/en-us/azure/data-factory/connector-overview), such as Blob, S3, GCS, and Big Query. It would be advantageous to make the extension more flexible and able to support additional connectors.
- It would be a good practice that supports data transfer between different types of sources and destinations.

## Solution Proposal

I am doing the implementation now, please check the sequence diagram and decision record for detail.
[Blob to Dataverse Sequence Diagram](https://github.com/huanji-sun-007/Connector/blob/ee674c4c16a8f126b8ceaf842d195264e0898ca2/docs/developer/decision-records/2023-04-12-data-factory-support-blob-to-dataverse/blob-to-dataverse-sequence.png)
[Decision Record](https://github.com/huanji-sun-007/Connector/blob/ee674c4c16a8f126b8ceaf842d195264e0898ca2/docs/developer/decision-records/2023-04-12-data-factory-support-blob-to-dataverse/README.md)

**Comment by huanji-sun-007** [comment]
Please let me mention @paullatzelsperger @bscholtes1A @jimmarino (who worked the Data Factory [related PR](https://github.com/eclipse-edc/Connector/pull/910) before)
It would be really helpful if you can give us your opinions and feedback. Thank you in advance :)

**Comment by huanji-sun-007** [comment]
I confirmed with @jimmarino. The project committee is working on splitting the core connector repo.  [decision record repository split](https://github.com/eclipse-edc/Connector/tree/main/docs/developer/decision-records/2023-03-15_repository_split). Cloud-specific features will be split into separate repos. So at this moment, it's not good timing to merge new Azure-specific features into Connecor repo. 

I pushed the implementation of `Support data transfer from blob storage to Dataverse by using data-factory extension` in my [forked repo](https://github.com/huanji-sun-007/Connector/pull/1) for now.  Will consider merge this feature to the `Technology-Azure` in the future.
